src/GridFS/Bucket.php

```diff
@@ -644,6 +646,7 @@ public function uploadFromStream(string $filename, $source, array $options = [])
     private function createPathForFile(object $file): string
     {
         if (! is_object($file->_id) || method_exists($file->_id, '__toString')) {
+            /** @psalm-suppress PossiblyInvalidCast */
```

Comment:
The full error is:
> ERROR: PossiblyInvalidCast - src/GridFS/Bucket.php:649:28 - object cannot be cast to string (see https://psalm.dev/190)

I rewrote the conditional to avoid the error:
```php
if (is_array($file->_id) || (is_object($file->_id) && ! method_exists($file->_id, '__toString'))) {
    $id = toJSON(fromPHP(['_id' => $file->_id]));
} else {
    $id = (string) $file->_id;
}
```

I believe this covers all cases: arrays and objects without a `__toString` method are passed to `fromPHP`, while all other values are cast to string.

Reply:
Oh, so this has nothing to do with an array. Psalm just had trouble understanding the inverted conditional. SGTM.